typings/contacthub-sdk-nodejs.d.ts

```diff
@@ -0,0 +1,477 @@
+  registeredAt?: string,
+  updatedAt?: string,
+  preferences: Array<Preference>
+};
+
+type APIBaseProperties = {
+  pictureUrl?: string,
+  title?: string,
+  prefix?: string,
+  firstName?: string,
+  lastName?: string,
+  middleName?: string,
+  gender?: string,
+  dob?: string,
+  locale?: string,
+  timezone?: string,
+  contacts?: APIContacts,
+  address?: APIAddress,
+  credential?: APICredential,
+  educations: Array<APIEducation>,
+  likes: Array<APILike>,
+  socialProfile: APISocial,
+  jobs: Array<APIJob>,
+  subscriptions: Array<APISubscription>
+};
+
+type APICustomerData = {
+  externalId?: string,
+  extra?: string,
+  base?: APIBaseProperties,
+  extended?: Object,
+  tags?: Tags
+};
+
+type APICustomer = APICustomerData & {
+  id: string,
+  nodeId: string,
+  enabled: boolean,
+  registeredAt: string,
+  updatedAt: string
+};
+
+type EventFilters = {
+  type?: EventType,
+  context?: EventContext,
+  mode?: 'ACTIVE' | 'PASSIVE',
+  dateFrom?: Date,
+  dateTo?: Date,
+  page?: number
+};
+
+type Paginated<T> = {
+  page: {
+    current: number,
+    total: number,
+    prev?: () => Promise<Paginated<T>>,
+    next?: () => Promise<Paginated<T>>
+  },
+  elements: Array<T>
+};
+
+declare class ContactHub {
+  constructor(params: Auth)
+
+  createSessionId(): string;
+
+  addCustomerSession(customerId: string, sessionId: string): Promise<boolean>;
+
+  addCustomer(customerData: CustomerData): Promise<Customer>;
+
+  getCustomer(customerId: string): Promise<Customer>;
+
+  getCustomers(options?: GetCustomersOptions): Promise<Paginated<Customer>>;
+
+  updateCustomer(customerId: string, customerData: CustomerData): Promise<Customer>;
+
+  patchCustomer(customerId: string, customerData: CustomerData): Promise<Customer>;
+
+  deleteCustomer(customerId: string): Promise<boolean>;
+
+  addEducation(customerId: string, education: Education): Promise<Education>;
+
+  updateEducation(customerId: string, education: Education): Promise<Education>;
+
+  deleteEducation(customerId: string, educationId: string): Promise<boolean>;
+
+  addJob(customerId: string, job: Job): Promise<Job>;
+
+  updateJob(customerId: string, job: Job): Promise<Job> ;
+
+  deleteJob(customerId: string, jobId: string): Promise<boolean>;
+
+  addLike(customerId: string, like: Like): Promise<Like>;
+
+  updateLike(customerId: string, like: Like): Promise<Like>;
+
+  deleteLike(customerId: string, likeId: string): Promise<boolean>;
+
+  addEvent(event: EventData): Promise<boolean>;
+
+  getEvent(eventId: string): Promise<ContactHubEvent>;
+
+  getEvents(customerId: string, filters?: EventFilters): Promise<Paginated<ContactHubEvent>>;
+
+  addTag(customerId: string, tag: string): Promise<Customer>;
+
+  removeTag(customerId: string, tag: string): Promise<Customer>;
+
+  createQuery(attribute: string, operator: AtomicConditionOperator, value?: any): Query;
+}
+
+declare module 'contacthub-sdk-nodejs' {
+  export = ContactHub;
+  namespace ContactHub {}
+}
```

Comment:
I didn't review this file in detail. Is it just because the TS PR on contacthub-sdk-nodejs is not merged/released yet?